### ISSUE TITLE
508-defect-3 [FOCUS MANAGEMENT]: Focus outline SHOULD be removed from non-linked heading

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#8279]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

When a user submits criteria for a forms search, focus moves to the "Showing 1–10 of 13 results for "medical"" heading. The yellow halo **should** be removed from the search results string, as focus outlines should be reserved for interactive items. This is a low-priority item and ultimately should be captured in our VA.gov design guide.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a keyboard user, I want to identify interactive focusable items by distinctive focus styles.

## Environment

* Operating System: all
* Browser: all
* Server destination: production

## Steps to Recreate

1. Enter https://www.va.gov/find-forms?q=medical in browser
2. Verify the focus outline is on the heading indicating the number of search results found

## Possible Fixes (optional)

See https://github.com/department-of-veterans-affairs/va.gov-team/issues/457

## WCAG or Vendor Guidance (optional)

* [W3C Understanding WCAG 2.0 - Consistent Identification SC 3.2.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-consistent-functionality.html)

## Screenshots or Trace Logs

![Screen Shot 2020-04-22 at 1 03 49 PM](https://user-images.githubusercontent.com/57469/80012133-d44f0580-849a-11ea-912f-1f4f31035a23.png)

**Comment by jenstrickland** [comment]
@1Copenut Reviewing this ticket, I wonder… on Facility Locator, when a user submits search criteria we move focus to the `h2` that announces the search results, and we have a yellow focus indicator. It seems this behavior is actually consistent with the pattern we use across our work. What do you think?

**Comment by 1Copenut** [comment]
@jenstrickland The yellow focus halo here is probably the default `::focus` rule being picked up. I'd opt for it to be removed here in the name of consistency.

**Comment by jenstrickland** [comment]
@1Copenut Should we also remove it here? 

https://www.va.gov/find-locations/?zoomLevel=4&page=1&address=lincoln%2C%20nebraska&facilityType=health&serviceType&location=40.8001%2C-96.6674&context=Lincoln%2C%20Nebraska%2C%20United%20States

**Comment by MarciMcGuireGCIO** [comment]
Moving all Find a VA Form 508 issues to a single epic to clean up backlog.

**Comment by MarciMcGuireGCIO** [comment]
@joshkimux would you mind reviewing this when you have a few minutes to see if it's still valid?  Our concern is that removing the tag as recommended would cause different issues.

**Comment by MarciMcGuireGCIO** [comment]
@joshkimux Yayyyyyyyy!  

Would it be better for us to have a standing request each week/sprint as Zach is pulling 508 issues from our backlog to resolve? I hate to keep just pinging you ad-hoc if there's a better way to manage.  Thanks! 😸

**Comment by joshkimux** [comment]
@MarciMcGuireGCIO That would be awesome!

**Comment by MarciMcGuireGCIO** [comment]
@zacharymorel - please see above.  Any thought as to the best day for us to grab some of Josh's time for ongoing 508 reviews?

**Comment by zacharymorel** [comment]
I'm good with Monday's or Friday's.

**Comment by joshkimux** [comment]
Mondays are best for me!

**Comment by joshkimux** [comment]
<img width="923" alt="Screen Shot 2021-03-11 at 9 09 30 AM" src="https://user-images.githubusercontent.com/14154792/110799756-7f021b00-8249-11eb-9e4f-4b84a4e73e8e.png">

@MarciMcGuireGCIO , just did a quick test and the halo is still present. I understand the original concern is important as a focus halo may imply that it can be interacted with-- particularly to users with assistive technology.

In this case, [toggling the focus style should do the trick without any damage as documented in this article](https://css-tricks.com/focus-styles-non-interactive-elements/#but-but-but).

e.g.
```css
h2:focus, {
  outline: 0;
}
```

@1Copenut keep me honest here! I'm not a focus wizard like you 🙇

**Comment by MarciMcGuireGCIO** [comment]
Thanks, @joshkimux!  @1Copenut if we can also get your opinion before @zacharymorel makes any changes that would be awesome.  Sometimes the focus and accessibly rules seem a bit contradictory.  We just want to do what works best all around. Thanks!

**Comment by 1Copenut** [comment]
@MarciMcGuireGCIO Great question. We've tried to reduce our rules for focus halo to two broad declarations:

1. Show the focus halo on interactive elements that have keyboard focus. This includes buttons, links, form inputs, radio buttons, checkboxes.
2. Remove the focus halo when we set focus programmatically (usually with JavaScript) on non-native focusable elements. This mostly applies to headings for us when search results come back or we press "Continue" in a form.

I discussed this with our contact at the VA 508 office on another project. I the case that it could be more confusing for the yellow halo to be present on elements that normally wouldn't be focusable. Users can't tab to the H2--focus is only set when they search or update their search.

Happy to chat more about this anytime!

**Comment by zacharymorel** [comment]
@joshkimux This is live on staging, can you give it a look?

**Comment by joshkimux** [comment]
@zacharymorel Confirmed! Great work! 😄

**Comment by MarciMcGuireGCIO** [comment]
Documenting fix.

Before change:
![image](https://user-images.githubusercontent.com/73354907/113153629-49779e80-9205-11eb-9bf7-0903ab05bf44.png)


After change (no focus on/yellow box around the number of results):
![image](https://user-images.githubusercontent.com/73354907/113153769-6a3ff400-9205-11eb-9cd0-2debce1f07b8.png)
